typing/typemod.ml

```diff
@@ -889,11 +969,22 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
-        | Psig_open sod ->
-            let (_path, newenv, od) = type_open env sod in
-            let (trem, rem, final_env) = transl_sig newenv srem in
-            mksig (Tsig_open od) env loc :: trem,
-            rem, final_env
+        | Psig_open sod -> begin
```

Comment:
Please put the begin...end around the pattern matching (`match id ...`) instead.